src/Security/OAuthUserProvider.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the user bundle package.
+ * (c) Connect Holland.
+ */
+
+namespace ConnectHolland\UserBundle\Security;
+
+use ConnectHolland\UserBundle\Entity\User;
+use ConnectHolland\UserBundle\Entity\UserOAuth;
+use ConnectHolland\UserBundle\Repository\UserRepository;
+use Doctrine\Bundle\DoctrineBundle\Registry;
+use Doctrine\Common\Persistence\ObjectManager;
+use HWI\Bundle\OAuthBundle\OAuth\Response\UserResponseInterface;
+use HWI\Bundle\OAuthBundle\Security\Core\User\OAuthAwareUserProviderInterface;
+
+class OAuthUserProvider implements OAuthAwareUserProviderInterface
+{
+    /**
+     * @var Registry
+     */
+    private $doctrine;
+
+    public function __construct(Registry $doctrine)
+    {
+        $this->doctrine = $doctrine;
+    }
+
+    public function loadUserByOAuthUserResponse(UserResponseInterface $response)
+    {
+        /** @var UserRepository $repository */
+        $repository = $this->doctrine->getRepository(User::class);
+
+        $name     = $response->getResourceOwner()->getName();
+        $username = $response->getUsername();
+        $email    = $response->getEmail();
+
+        $user = $repository->findOneByOAuthUsername($name, $username);
+
+        if (is_null($user)) {
+            $user = $repository->findOneByEmail($email);
+
+            if (is_null($user)) {
+                $user = new User();
+                $user->setEnabled(true);
+                if ($email) {
+                    $user->setEmail($email);
+                }
+            }
+
+            $oauth = new UserOAuth();
+            $oauth->setResource($name);
+            $oauth->setOAuthUsername($username);
+
+            $user->addOAuth($oauth);
+
+            // @todo: Dispatch new UserOAuth created here to be able adding more data from $response.
+        }
+
+        $roles   = $user->getRoles();
+        $roles[] = 'ROLE_OAUTH';
```

Comment:
I'm not sure about this... Is this a role? This is history which made someone ROLE_USER, or ROLE_ADMIN.
I think you should add some mapping (and an event) to get the role and default to ROLE_USER.

Reply:
The events for this kind of places are on the roadmap, this seems sensible defaults for now

Reply:
I disagree to have something that is not role-related in your role name. You are not a blogger because OAUTH_GOOGLE...